Source/DafnyLanguageServer/Handlers/DafnyHoverHandler.cs

```diff
@@ -122,7 +125,7 @@ class AssertionBatchIndexComparer : IComparer<AssertionBatchIndex> {
       }
 
       var assertionBatchesToReport =
-        node.AssertionBatches.Values.OrderByDescending(a => a.ResourceCount).Take(3).ToList();
+        orderedAssertionBatches.OrderByDescending(a => a.ResourceCount).Take(3).ToList();
```

Comment:
Oh nice finding. Indeed, it could be that between the two the list had grown !

Reply:
I think it would be better to make an immutable copy of relevant information from `VerificationTree` in `DocumentAfterTranslation.ToIdeState`, which downstream consumers like this hover handler can then use.